src/pudl_rmi/connect_deprish_to_ferc1.py

```diff
@@ -66,762 +191,380 @@ def execute(plant_parts_eia, deprish_eia, ferc1_to_eia, clobber=False):
-        """
-        self.scale_by_ownership_fraction()
-        same_true = self.assign_ferc1_data_cols_same()
-        same_smol = self.split_ferc1_data_cols()
-        same_beeg = self.agg_ferc_data_cols()
-        logger.info(f"Scaled via same/true: {len(same_true)}")
-        logger.info(f"Scaled via same/smol: {len(same_smol)}")
-        logger.info(f"Scaled via same/beeg: {len(same_beeg)}")
-        scaled_df = pd.concat([same_true, same_smol, same_beeg])
-
-        self.test_same_true_fraction_owned(same_true)
-        logger.info(
-            f"output is len {len(scaled_df)} while input was "
-            f"{len(self.matches_df)}"
+            to_allocate: the dataframe to be allocated.
+            ppl: EIA plant-part list generated from
+                ``pudl.analysis.plant_parts_eia``
+        """
+        # STEP 2
+        merged_df = self.many_merge_on_scale_part(
+            to_scale=to_allocate,
+            ppl=ppl.reset_index(),
+            cols_to_keep=list(self.treatments)
         )
-        return scaled_df
-
-    def scale_by_ownership_fraction(self):
-        """
-        Scale the data columns by the fraction owned ratio in matches_df.
-
-        This method makes new columns in matches_df for each of the data
-        columns in `DATA_COLS_TO_SPLIT` that scale the ferc1 data based on the
-        ownership fractions of the depreciation and ferc1 records.
-        """
-        for data_col in DATA_COLS_TO_SPLIT:
-            self.matches_df.loc[:, f"{data_col}_own_frac"] = (
-                self.matches_df[data_col]
-                * (self.matches_df.fraction_owned_deprish
-                   / self.matches_df.fraction_owned_ferc1)
-            )
-
-    def assign_ferc1_data_cols_same(self):
-        """
-        Assign FERC1 data cols to deprecation when records are the same.
-
-        For matched depreciation and FERC1 records are exactly the same, this
-        method simply assigns the original data column to the new associated
-        data column (with the format of "{data_col}_ferc1_deprish"). This
-        method scales ferc1 data for both the `same_true` match method and the
-        `same_diff_own` method - it uses the scaled `{data_col}_own_frac`
-        generated in `scale_by_ownership_fraction()`.
-
-        Relies on:
-        * matches_df (pandas.DataFrame): a dataframe of records from the
-            depreciation data with ferc1 steam data associated in a many to
-            many relationship.
-        """
-        same_true = deepcopy(
-            self.matches_df.loc[
-                (self.matches_df.match_method == 'same_true')
-                | (self.matches_df.match_method == 'same_diff_own')
-            ]
+        # at this stage we have plant-part records with the data columns from
+        # the original dataset broadcast across multiple records: read the data
+        # is duplicated!
+
+        # STEP 3
+        # grab all of the ppl columns, plus data set's id column(s)
```

Comment:
When you say "id column" are you referring to the "index columns" and not the "identifier columns?" To me ID just means identifier, not index.

Reply:
I've now standardized the way i refer to ID columns. I mean identifier, not index. I've gone back and forth calling them primary key columns, because usually they would be the PKs of a table.

Reply:
scratch that. I am now calling them PK columns and defined PK in the module docs